File: cocos2d-x-2.1.4/extensions/CCBReader/AceScale9SpriteNode.cpp
//
//  AceScale9SpriteNode.cpp
//  MLB
//
//  Created by 허정목 on 2014. 5. 23..
//
//

#include "AceScale9SpriteNode.h"

USING_NS_CC_EXT;

void AceScale9SpriteNode::onEnter()
{
    CCNode::onEnter();
    
    CCArray* children = this->getChildren();
    int size = children->count();
    for(int i=0; i<size; i++)
    {
        CCScale9Sprite* pNode = (CCScale9Sprite*)children->objectAtIndex(i);

        if(pNode)
        {
            pNode->setContentSize(this->getContentSize());
            
        }

    }
    
}

File: CocosBuilder-3.0.alpha5-src/CocosBuilder/AcePlugIns/AceScale9SpriteNode/AceScale9SpriteNode.m
/*
 * CocosBuilder: http://www.cocosbuilder.com
 *
 * Copyright (c) 2012 Zynga Inc.
 *
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in
 * all copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 * THE SOFTWARE.
 */

#import "AceScale9SpriteNode.h"
#import "CCScale9Sprite.h"

@implementation AceScale9SpriteNode



- (id) init
{
    self = [super init];
    if (!self) return NULL;
    

    
    return self;
}

-(void) visit
{
    CCArray* children = [self children];
    int size = (int)children.count;
    
    for(int i=0; i<size; i++)
    {
        CCScale9Sprite* pNode = (CCScale9Sprite*)[children objectAtIndex:(int)i];
        if(pNode)
        {
            [pNode setContentSize:[self contentSize]];
        }
    }
    
    [super visit];
}


@end
